### PR TITLE
MGMT-2281 OCM Client connection reuse

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by rollandf** [comment]
@carbonin

**Comment by rollandf** [comment]
/hold
Need to verify with test-infra with Auth

**Comment by romfreiman** [comment]
@tsorya

File: pkg/ocm/client.go [hunk 1]
```diff
@@ -99,14 +103,16 @@ func (c *Client) NewConnection() (*sdkClient.Connection, error) {
...
 	} else {
-		return nil, fmt.Errorf("Can't build OCM client connection. No Client/Secret or Token has been provided")
+		return fmt.Errorf("Can't build OCM client connection. No Client/Secret or Token has been provided.")
 	}
 
 	connection, err := builder.Build()
+
 	if err != nil {
-		return nil, fmt.Errorf("Can't build OCM client connection: %s", err.Error())
+		return fmt.Errorf("Can't build OCM client connection: %s", err.Error())
```
Comment (tsorya) [inline pkg/ocm/client.go:112]: errors.Wrap is better

File: pkg/ocm/client.go [hunk 2]
```diff
@@ -77,6 +78,10 @@ func NewClient(config Config, log logrus.FieldLogger) (*Client, error) {
 		logger: logger,
 		Cache:  cache.New(1*time.Minute, 30*time.Minute),
 	}
+	err := client.newConnection()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to build OCM connection: %s", err.Error())
```
Comment (tsorya) [inline pkg/ocm/client.go:83]: Why not errors.Wrap? i think we mostly use it.

**Comment by tsorya** [comment]
@rollandf u need to rebase

File: pkg/ocm/client.go (hunk 1 above)
Comment (rollandf) [reply pkg/ocm/client.go]: Done

File: pkg/ocm/client.go (hunk 2 above)
Comment (rollandf) [reply pkg/ocm/client.go]: Done

**Review by carbonin — APPROVED** [review]

**Review by filanov — COMMENTED** [review]
What about connection renew? if we get disconnected do we need to renew the connection? otherwise all the apis will fail

**Comment by rollandf** [comment]
> What about connection renew? if we get disconnected do we need to renew the connection? otherwise all the apis will fail

There is no need to renew. This is not a TCP connection that is always open. It is more of an helper that is handling SSO token and holding the needed parameters for sending HTTP requests when needed.

File: pkg/ocm/client.go [hunk 3]
```diff
@@ -99,14 +104,16 @@ func (c *Client) NewConnection() (*sdkClient.Connection, error) {
 	} else if c.config.SelfToken != "" {
 		builder = builder.Tokens(c.config.SelfToken)
 	} else {
-		return nil, fmt.Errorf("Can't build OCM client connection. No Client/Secret or Token has been provided")
+		return fmt.Errorf("Can't build OCM client connection. No Client/Secret or Token has been provided.")
```
Comment (filanov) [inline pkg/ocm/client.go:107]: please use the errors package 
```suggestion
		return errors.Errorf("Can't build OCM client connection. No Client/Secret or Token has been provided.")
```
it have a stuck trace option.

Comment (rollandf) [reply pkg/ocm/client.go]: Done

**Review by filanov — APPROVED** [review]

**Pull request merged**